### ISSUE TITLE
Implement the setting up of a new battle for Battle Flow module

### ISSUE THREAD

**Issue by celiaanderson** [uchicago-cs/chiventure#534]
Write an implementation of functions that would occur at the beginning of a new battle, such as setting up structs for the battle, in the Battle Flow module, according to the specifications in the Battle Systems design document, as we are beginning to add code that follows the updated design modules. This implementation will help with preparing for an implementation of the actual step by step runtime battle flow in the rest of the Battle Flow module (issue #537)

**Comment by celiaanderson** [comment]
Created a new branch to implement battle flow (battle/battle-flow). Added header and c files and started on implementations for set_battle, set_player, set_enemies, and start_battle functions. Originally was going to have a separate function just for setting up all the structs of the battle, but it would be simpler in terms of passing around variables to have set_battle, set_player, and set_enemies at the beginning of the start_battle, which will be the main function for the battle flow that sets up the battle and then runs the battle loop.  

Additionally, since battle flow is at a higher level, I wanted to include some more direct references to the rest of the chiventure implementation in addition to the rest of the battle modules, so I have been looking at working with chiventure_context_t and some of the ui functions. Once I get a more complete implementation of the battle set-up for combatant and battle structs, I want to try out writing a function that would provide the message for a new battle. I also created a struct to encompass all the stubs for stats, items, and moves to make things a little more concise for now and made a npc_t stub for how the enemy NPCs might be passed into the battle.

**Comment by celiaanderson** [comment]
The first pass at implementing the battle-setup has been done, complete with tests in battle_flow_test.c, and I have put up a pull request. 
Some changes from my previous approach: rather than referencing chiventure_ctx_t, player_t, etc directly, I've made stubs to simulate those structs and relevant functions - partly to avoid clashing with the current item_t stub in battle_structs/battle_state, and partly so that I could also include a new field in a chiventure_ctx_t stub that holds a boolean for if there is a battle going on. 
Currently, the functions for setting up a new battle continue to be set_player, set_enemies, and set_battle. set_battle() will call set_player and set_enemies to set up the combatant structs for the player and NPC enemies respectively. Then, start_battle(), which will later also hold the loop for running the battle, calls set_battle() to make a new battle given information about the environment, player and enemies. Right now, start_battle() itself takes in the ctx struct (which would hold the current player that is engaging in a battle), a list of NPC enemies, and the environment to battle in. Presumably, the environment parameter in the future may be accessible through looking at the game struct within the ctx struct.

**Comment by celiaanderson** [comment]
Currently in the process of getting the pull request approved (https://github.com/uchicago-cs/chiventure/pull/623). Some changes that have happened so far include changing the battle_mode field in the chiventure context stub to be called in-battle, where the in-battle field marks whether or not a battle is in progress. I added tests for fighting one enemy in addition to my test for two enemies. I moved my stubs to battle_flow_structs.c/.h with tests for making a new player_t stub, npc_enemy_t stub, and game_t stub. The latest changes are at https://github.com/uchicago-cs/chiventure/pull/623/commits/b5859572f84182461c7b327f29445e9cfb233e73.

**Comment by celiaanderson** [comment]
Closing this issue - the pull request (https://github.com/uchicago-cs/chiventure/pull/623) has been approved and battle flow set up has been merged. Final changes (https://github.com/uchicago-cs/chiventure/pull/623/commits/89bd5a2a823e1b770dc557d986e4eb51a24e1ffb) were to add some includes, remove duplicate structs, and changing the ifndef label for battle structs. In summary, I have completed and tested (resulting in merging to dev) an implementation of setting up a new battle with functions start_battle, set_player, set_enemies, and set_battle (for setting up the player/enemy with combatant_t structs, and setting up a new battle with battle_t using the player, enemy, and environment information) that utilizes the battle_state module and stubs that simulate interacting with chiventure context, game_t, player_t, and an NPC struct separate from battle-specific structs.
